daemon/config/config.go

```diff
@@ -71,6 +72,12 @@ type commonBridgeConfig struct {
 	FixedCIDR string `json:"fixed-cidr,omitempty"`
 }
 
+// NetworkConfig stores the daemon-wide networking configurations
+type NetworkConfig struct {
+	// Default address pools for docker networks
+	DefaultAddressPools []*ipamutils.NetworkToSplit `json:"default-node-local-address-pools,omitempty"`
```

Comment:
Now I see why you were using a pointer to a slice.

The type here can be `PoolsOpt` instead of `[]*ipamutils.NetworkToSplit`.

That way you don't need to accept any value in `NewPoolsOpt`, and to get the value out you can use: `DefaultAddressPools.Value()`